src/ObjectUtils/PdbFileLlvm.cpp

```diff
@@ -214,6 +255,118 @@ PdbFileLlvm::PdbFileLlvm(std::filesystem::path file_path, const ObjectFileInfo&
                           llvm::toString(std::move(error)))};
     }
   }
+  return outcome::success();
+}
+
+void LoadDebugSymbolsFromPublicSymbolStream(
+    llvm::pdb::PublicsStream& public_symbol_stream, llvm::pdb::SymbolStream& symbol_stream,
+    llvm::FixedStreamArray<llvm::object::coff_section>& section_headers,
+    const ObjectFileInfo& object_file_info,
+    const absl::flat_hash_set<uint64_t>& addresses_from_module_debug_stream,
+    std::vector<SymbolInfo>* symbol_infos) {
+  const llvm::pdb::GSIHashTable& public_symbol_has_records = public_symbol_stream.getPublicsTable();
+  for (const auto& hash_record : public_symbol_has_records) {
+    llvm::Expected<llvm::codeview::PublicSym32> record =
+        llvm::codeview::SymbolDeserializer::deserializeAs<llvm::codeview::PublicSym32>(
+            symbol_stream.readRecord(hash_record));
+    ORBIT_CHECK(record);
+
+    // Skip this symbol if it is not a function (but rather a global constant).
+    if ((record->Flags & llvm::codeview::PublicSymFlags::Function) ==
+        llvm::codeview::PublicSymFlags::None) {
+      continue;
+    }
+
+    uint64_t address = ComputeAddress(record->Offset, record->Segment, object_file_info.load_bias,
+                                      section_headers);
+
+    if (addresses_from_module_debug_stream.contains(address)) continue;
+
+    SymbolInfo symbol_info;
+    symbol_info.set_address(address);
+    symbol_info.set_demangled_name(llvm::demangle(record->Name.str()));
+    // The PDB public symbols don't contain the size of symbol. Set a placeholder which indicates
+    // that the size is unknown for now and try to deduce it later. We will later use that
+    // placeholder in DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol.
```

Comment:
This comment is partly obsolete because now we also use `SectionContributionsVisitor` (before `DeduceDebugSymbolMissingSizesAsDistanceFromNextSymbol`) to get the size of symbols with `kUnknownSymbolSize` set.